fast64_internal/oot/oot_actor.py

```diff
@@ -1,47 +1,410 @@
+					if isinstance(value, bool):
+						if value: param = '0x1'
+						else: param = '0x0'
+					else:
+						param = value
+					setActorString(elem, eval(param), self, dPKey, lenProp, lenSwitch, lenBool, lenEnum, paramTarget)
+
+def genEnum(annotations, key, suffix, enumList, enumName):
+	'''This function is used to generate the proper enum blender property'''
+	objName = key + suffix
+	prop = bpy.props.EnumProperty(name=enumName, items=enumList)
+	annotations[objName] = prop
+
+def genString(annotations, key, suffix, stringName):
+	'''This function is used to generate the proper string blender property'''
+	objName = key + suffix
+	prop = bpy.props.StringProperty(name=stringName, default='0x0')
+	annotations[objName] = prop
+
+def drawParams(box, detailedProp, key, elemField, elemName, elTag, elType, lenSwitch):
+	'''Actual draw on the UI'''
+	for actorNode in root:
+		i = 1
+		name = 'None'
+		if key == actorNode.get('Key'):
+			for elem in actorNode:
+				# Checks if there's at least 2 Switch Flags, in this case the
+				# Name will be 'Switch Flag #[number]
+				# If it's not a Switch Flag, change nothing to the name
+				if lenSwitch is not None:
+					if int(lenSwitch) > 1: name = f'{elemName} #{i}'
+					else: name = elemName
+				else: name = elemName
+
+				# Set the name to none to use the element's name instead
+				# Set the name to the element's name if it's a flag and its name is set
+				curName = elem.get('Name')
+				if elemName is None or (elTag == 'Flag' and curName is not None): name = curName
+
+				# Add the index to get the proper attribute
+				if elTag == 'Property' or (elTag == 'Flag' and elType == 'Switch') or elTag == 'Bool' or elTag == 'Enum':
+					field = elemField + f'{i}'
+				else:
+					field = elemField
+
+				attr = getattr(detailedProp, field, None)
+				tiedParam = elem.get('TiedParam')
+				actorType = getattr(detailedProp, key + '.type', None)
+				if name != 'None' and elem.tag == elTag and elType == elem.get('Type') and attr is not None:
+					if hasTiedParams(tiedParam, actorType) is True:
+						prop_split(box, detailedProp, field, name)
+					i += 1
+
+def drawOperatorBox(layout, obj, detailedProp, field, labelText, searchOp, enum):
+	searchOp.objName = obj
+	split = layout.split(factor=0.5)
+	split.label(text=labelText)
+	split.label(text=getEnumName(enum, getattr(detailedProp, field)))
+
+def editOOTActorProperties():
+	'''This function is used to edit the OOTActorProperties class before it's registered'''
+	propAnnotations = getattr(OOTActorProperties, '__annotations__', None)
+	if propAnnotations is None:
+		propAnnotations = {}
+		OOTActorProperties.__annotations__ = propAnnotations
+
+	# Each prop has its 'custom' variant because of the get and set functions
+	# Actors/Entrance Actor
+	propAnnotations['actorID'] = bpy.props.EnumProperty(name='Actor ID', items=ootEnumActorID)
+	propAnnotations['actorIDCustom'] = bpy.props.StringProperty(name='Actor Key', default='ACTOR_CUSTOM')
```

Comment:
imo these properties that are not dynamic (don't depend on the xml data) can be in `OOTActorProperties` directly

Reply:
I have some issues when I move things like ``actorParam``

Reply:
I can't move a lot of annotations since some are needed for the upgrade, I think I can only move the ``ToSave`` annotations since it's only used for saving in the blend